feat(docs): Add documentation for "attached service" notification events

rfk opened the pull request:
@vladikoff as mentioned earlier in the week, here are my initial notes on the events we publish and what services should do when they receive them.  r?

vbudhram commented on docs/service_notifications.md:
```diff
@@ -0,0 +1,168 @@
+The types of event, the additional data they contain,
+and the actions that attached services should take in response
+are as follows.
+
```
I think it might be worthwhile to add here that all messages have a `ts` (seconds float) and can optional have a `metricsContext` payload.

rfk commented:
I've rebased this, and added the new event introduced in https://github.com/mozilla/fxa-auth-server/pull/2390 on the presumption that it will be uncontroversial.

vladikoff commented on docs/service_notifications.md:
```diff
@@ -0,0 +1,199 @@
+when a user deletes their Firefox Account.
+
+To help services meet these requirements,
+FxA publishes a stream of SQS notification events.
```
`SQS` can we link out to SQS docs or explanation of what that is?

vladikoff commented on docs/service_notifications.md:
```diff
@@ -0,0 +1,199 @@
+
+Sync is the only service
+that currently knows about connected devices,
+so other services probably shouldn't use this.
```
Can we link to examples of where we consume these events? I think we have a node.js and a Python example in the wild.

vladikoff commented on docs/service_notifications.md:
```diff
@@ -0,0 +1,199 @@
+as a consumer of this stream,
+please email fxa-staff@mozilla.com for assistance.
+
+Each notification event has a stringified JSON body
```
nit: add an example of the body of the notification , just a nit though

rfk commented:
Thanks @vladikoff, updated and will merge when travis goes green.
